src/queue/Queue.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * The queue package contains our Queue, QueueTest, QueueNode and QueueTest 
+ * classes.
+ * 
+ * @author Scott McDonald and Jon Cronin
+ */
+package queue;
+
+/**
+ * Queue implements the basic Queue ADT, with the functions enque() and deque().
+ * Users of this queue class can set the maximum capacity or accept the default
+ * of 10.
+ * @author Scott McDonald and Jon Cronin
+ */
+public class Queue {
+    private QueueNode head;
+    private QueueNode tail;
+    private int capacity;
+    private int size;
+    
+    // Define the return values the Queue class methods may use
+    public static final int Q_SUCCESS = 0;
+    public static final int Q_ERR_NULL = -1;
+    public static final int Q_ERR_FULL = -2;
+    
+    /**
+     * Default constructor. Sets head and tail to null, max capacity to 10 and
+     * current size to 0.
+     */
+    public Queue() {
+        head = null;
+        tail = null;
+        capacity = 10;
+        size = 0;
+    }
+    
+    /**
+     * Alternate constructor that allows caller to set the maximum capacity.
+     * If capacity is set to a negative value then we will simply let the queue
+     * grow until the system is out of heap.
+     * @param capacity 
+     */
+    public Queue(int capacity) {
+        this();
```

Comment:
that calls the default Queue constructor from the overloaded constructor so you don't have to rewrite all the lines and only have to worry about the differences between the constructor calls. You can do the same thing with inherited classes and their constructors but then you use super() instead of this(). this is shorthand for this object instance, so from the object instance you are accessing the object instance.